test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -277,7 +278,22 @@ func TestApplyResetFields(t *testing.T) {
 						Namespace(namespace).
 						Patch(context.TODO(), name, types.ApplyPatchType, obj2YAML, metav1.PatchOptions{FieldManager: "fieldmanager2"}, "")
 					if err == nil || !strings.Contains(err.Error(), "conflict") {
-						t.Fatalf("expected conflict, got error %v", err)
+						var gotten string
+						obj3, err3 := dynamicClient.
+							Resource(mapping.Resource).
+							Namespace(namespace).
+							Get(context.TODO(), name, metav1.GetOptions{})
+						if err3 == nil {
+							marshBytes, marshErr := json.Marshal(obj3)
```

Comment:
Why do we need to marshal? Why not simply log `"%#v", obj3` ?

Reply:
I think the suggested replacement elides deeply nested content.

Reply:
The patch command is returning the object (L276) exactly as it was after the patch was applied, that's less likely to be racy

Reply:
I expect that if `err != nil` then the object is not returned.  Is that right?

Reply:
@apelisse , what is less likely to be racy than what?

Reply:
"Doing a GET after applying the PATCH" is more racy than "Using the object returned with the PATCH"

Reply:
> "Doing a GET after applying the PATCH" is more racy than "Using the object returned with the PATCH"

@apelisse That is a good point.
The following integration code seems to have the similar issue. Do we need to change it?

https://github.com/kubernetes/kubernetes/blob/e1d92980e35df3936f780930f4d6fde3fc23448a/test/integration/dualstack/dualstack_test.go#L1307-L1315

Sorry for bringing different test code here.